src/node/ws-master-http-client.ts

```diff
@@ -0,0 +1,85 @@
+/*********************************************************************
+ * Copyright (c) 2018 Red Hat, Inc.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ **********************************************************************/
+
+import { AxiosInstance, AxiosPromise, AxiosRequestConfig, default as axios } from 'axios'
+
+/**
+ * Plain wrapper around {@link AxiosInstance} class that is configured to
+ * have workspace API endpoint location as base URL. The value of API endpoint
+ * location is fetched out of environment variable called 'CHE_API'. If there
+ * is no such variable set the client tries to continue with default property
+ * values, however no correct request addressing is guaranteed in this case.
+ *
+ */
+export class WsMasterHttpClient {
+
+    private readonly ISSUE = 'Can\'t properly configure workspace master http client';
+    private readonly REASON = 'CHE_API environment property is undefined';
+
+    private readonly axiosInstance: AxiosInstance;
+
+    constructor(cheApi: string | undefined) {
+        if (cheApi) {
+            this.axiosInstance = axios.create({ baseURL: cheApi });
```

Comment:
have this been tested on che.openshift.io ?
what about machine token ?

Reply:
You are right, it will not work with multi-user che. But we have the same problem in `che-theia-ssh-plugin` so we need to find a common solution, but that's another issue.

Reply:
There are other plugins/extensions that works well with machine token. The common solution will be the CHE_API, but by the time, why not making it working ?

Reply:
fixed